landoapi/workers/landing_worker.py

```diff
@@ -291,26 +290,18 @@ def run_job(
                 self.notify_user_of_landing_failure(job)
                 return True
 
-            # Fetch all patches.
-            all_patches = [
-                (revision.revision_id, BytesIO(revision.patch_bytes))
-                for revision in job.revisions
-            ]
-
             # Run through the patches one by one and try to apply them.
-            for revision_id, patch_buf in all_patches:
+            for patch_id, patch_buf in job.get_patches():
```

Comment:
Upon rethinking this a little more, I think it would make more sense to access the revisions directly here.
```suggestion
            for revision in job.revisions:
                hgrepo.apply_patch(BytesIO(revision.patch_bytes))
```

Revisions are [lazily-loaded](https://docs.sqlalchemy.org/en/14/orm/loading_relationships.html#lazy-loading) anyway.

And then down the road, access the necessary variables directly, e.g.,

```python
breakdown = self.process_merge_conflict(exc, repo, hgrepo, revision.revision_id)
```

I don't think this breakdown is necessary any more, and is just a small remnant of the old implementation.